crates/move-prover-lean-backend/src/lean_backend/bytecode_translator.rs

```diff
@@ -0,0 +1,3224 @@
+                }
+            }
+        }
+        
+        ensures_info
+    }
+
+    /// Translates the given function.
+    fn translate(mut self) {
+        self.translate_with_ensures_control(true);
+    }
+
+    /// Translates one bytecode instruction.
+    fn translate_bytecode(
+        &mut self,
+        last_tracked_loc: &mut Option<(Loc, LineIndex)>,
+        bytecode: &Bytecode,
+    ) {
+        use Bytecode::*;
+
+        let spec_translator = &self.parent.spec_translator;
+        let options = self.parent.options;
+        let fun_target = self.fun_target;
+        let env = fun_target.global_env();
+
+        // Set location of this code in the CodeWriter.
+        let attr_id = bytecode.get_attr_id();
+        let loc = fun_target.get_bytecode_loc(attr_id);
+        self.writer().set_location(&loc);
+
+        // Print location.
+        emitln!(
+            self.writer(),
+            "-- {} {}",
+            bytecode.display(fun_target, &BTreeMap::default()),
+            loc.display(env)
+        );
+
+        // Print debug comments.
+        if let Some(comment) = fun_target.get_debug_comment(attr_id) {
+            emitln!(self.writer(), "-- {}", comment);
+        }
+
+        // Track location for execution traces.
+        if matches!(bytecode, Call(_, _, Operation::TraceAbort, ..)) {
+            // Ensure that aborts always has the precise location instead of the
+            // line-approximated one
+            *last_tracked_loc = None;
+        }
+        self.track_loc(last_tracked_loc, &loc);
+        if matches!(bytecode, Label(_, _)) {
+            // For labels, retrack the location after the label itself, so
+            // the information will not be missing if we jump to this label
+            *last_tracked_loc = None;
+        }
+
+        // Helper function to get a a string for a local
+        let str_local = |idx: usize| format!("t{}", idx);
+        let baseline_flag = self.fun_target.data.variant == FunctionVariant::Baseline;
+        let global_state = &self
+            .fun_target
+            .global_env()
+            .get_extension::<GlobalNumberOperationState>()
+            .expect("global number operation state");
+        let mid = self.fun_target.func_env.module_env.get_id();
+        let fid = self.fun_target.func_env.get_id();
+
+        // Translate the bytecode instruction.
+        match bytecode {
+            Call(_, dests, oper, srcs, aa) => {
+                use Operation::*;
+                match oper {
+                    FreezeRef => unreachable!(),
+                    UnpackRef | UnpackRefDeep | PackRef | PackRefDeep => {
+                        // No effect
+                    }
+                    OpaqueCallBegin(_, _, _) | OpaqueCallEnd(_, _, _) => {
+                        // These are just markers.  There is no generated code.
+                    }
+                    WriteBack(node, edge) => {}
```

Comment:
let's make up a macro, something like `wip!("WriteBack")`, that would print a comment in lean saying "WriteBack", to keep track of what's not implemented yet